Makefile

```diff
@@ -119,8 +121,17 @@ fmt: goimports golint
 .PHONY: check
 check: fmt test
 
-##@ Build
+##@ Build CLI Locally
+.PHONY: cli
+cli: $(CLI)
+$(CLI): $(LOCALBIN)
+	test -s $(LOCALBIN)/greenhousectl || echo "Building Greenhouse CLI..." && make build-cli
+
+.PHONY: build-cli
+build-cli:
+	go build -ldflags "-s -w -X github.com/cloudoperators/greenhouse/pkg/version.GitBranch=$(GIT_BRANCH) -X github.com/cloudoperators/greenhouse/pkg/version.GitCommit=$(GIT_COMMIT) -X github.com/cloudoperators/greenhouse/pkg/version.GitState=$(GIT_STATE) -X github.com/cloudoperators/greenhouse/pkg/version.BuildDate=$(BUILD_DATE)" -o bin/greenhousectl ./cmd/greenhousectl
```

Comment:
Can we do this instead? -

```
.PHONY: cli
cli: $(CLI)
$(CLI): $(LOCALBIN)
	test -s $(LOCALBIN)/greenhousectl || echo "Building Greenhouse CLI..." && make build-greenhousectl
```

That way when executing `make setup-dev` multiple times it will be faster as the cli already exists in bin and if it doesn't it will build it.

```shell
.PHONY: setup-dev
setup-dev: cli
	$(CLI) dev setup -f dev-env/localenv/sample.config.json
```